src/Microsoft.DotNet.Interactive.HttpRequest.Tests/ParserTests.cs

```diff
@@ -0,0 +1,645 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.DotNet.Interactive.HttpRequest.Tests.Utility;
+using Xunit;
+
+namespace Microsoft.DotNet.Interactive.HttpRequest.Tests;
+
+public class ParserTests
+{
+
+    public class Trivia
+    {
+        [Fact]
+        public void it_can_parse_an_empty_string()
+        {
+            var result = HttpRequestParser.Parse("");
+            result.SyntaxTree.Should().BeNull();
+
+            // TODO: Test error reporting.
+        }
+
+        [Fact]
+        public void it_can_parse_a_string_with_only_whitespace()
+        {
+            var result = HttpRequestParser.Parse(" \t ");
+
+            result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                  .MethodNode.ChildTokens.First().TextWithTrivia.Should().Be(" \t ");
+
+            // TODO: Test error reporting.
+        }
+
+        [Fact]
+        public void it_can_parse_a_string_with_only_newlines()
+        {
+            var result = HttpRequestParser.Parse("\r\n\n\r\n");
+
+            result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                  .MethodNode.TextWithTrivia.Should().Be("\r\n\n\r\n");
+
+            // TODO: Test error reporting.
+            // Regardless of number of newlines, there should be consistent nodes
+            // set of whitespace, newline, and punctuation
+            //each token kind
+            //combination of characters should be checked by the lexer
+        }
+    }
+    
+    
+    public class Lexer
+    {
+        [Fact]
+
+        public void multiple_whitespaces_are_treated_as_a_single_token()
```

Comment:
Please add similar tokenization tests for the other token kinds listed in the TODO comment above.

For example, punctuation should always be split into separate tokens, newline chars should always be split into separate tokens (except when it is `\r` followed by `\n` and words should always be combined into a single token.

The tokenization test for newlines would also address Jon's comment above.

Reply:
Looks like the tests for whitespace, newline and punctuation are now in place. It would be great to add one for "words should always be combined into a single token" so that we round out all the kinds of tokens that the lexer produces.